src/reservoir.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable import/no-unresolved */
+// @ts-nocheck
+import {
+  adaptViemWallet,
+  createClient,
+  getClient,
+  reservoirChains,
+} from "@reservoir0x/reservoir-sdk";
+import { Address, Hash, WalletClient, zeroAddress } from "viem";
+
+import { zeroHash } from "./blockchain";
+
+const ETH_CONTRACT_ADDRESS = zeroAddress;
+
+const generateExpectedCurrencyPriceObject = (
+  price: bigint,
+  currencyAddress: Address
+) => ({
+  [currencyAddress]: {
+    raw: price,
+    currencyAddress,
+    currencyDecimals: 18,
+  },
+});
+
+createClient({
+  chains: [
+    {
+      ...reservoirChains.mainnet,
+      active: true,
+    },
+  ],
+  // TODO: add api key
+  source: "gondi.xyz",
+});
+
+const adaptWalletToGetCallbackData = (wallet: WalletClient) => {
+  const callbackDataRef = { current: zeroHash };
+  const adaptedWallet = {
+    ...adaptViemWallet(wallet),
+    transport: undefined,
```

Comment:
This is ugly, I know. But it's the only way to make the reservoir sdk understand the wallet is already adapted. Otherwise, if transport is not undefined, it will re-adapt it with the viem wrapper. See: https://github.com/reservoirprotocol/reservoir-kit/blob/main/packages/sdk/src/utils/viemWallet.ts#L6